src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java

```diff
@@ -69,7 +69,13 @@ private boolean containsPropertiesTask(Settings settings) {
     private void configureBuildScan(Settings settings, GradleEnterpriseConventions conventions) {
         BuildScanExtension buildScan = settings.getExtensions().getByType(GradleEnterpriseExtension.class).getBuildScan();
 
-        buildScan.setServer(conventions.getGradleEnterpriseServerUrl());
+        if (conventions.getGradleEnterpriseServerUrl() == null) {
+            buildScan.setTermsOfServiceUrl("https://gradle.com/terms-of-service");
+            buildScan.setTermsOfServiceAgree("yes");
```

Comment:
I am not sure if we can agree for contributors to publish automatically to scans.gradle.com. Maybe you'd need to set an extra system property?